pkg/image/skopeo.go

```diff
@@ -95,6 +95,61 @@ func CopyRegistryImage(url, dest, accessKey, secKey string) error {
 	return err
 }
 
+// CopyDirFromRegistryImage download image from registry with skopeo and extracts a specific directory from it
+func CopyDirFromRegistryImage(url, dest, VMimagePath, accessKey, secKey string) error {
+	skopeoDest := "dir:" + dest + dataTmpDir
+	err := SkopeoInterface.CopyImage(url, skopeoDest, accessKey, secKey)
+	if err != nil {
+		os.RemoveAll(dest + dataTmpDir)
+		return errors.Wrap(err, "Failed to download from registry")
+	}
+	err = extractVMDiskImageDir(dest, VMimagePath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract image layers")
+	}
+
+	// Clean temp folder
+	os.RemoveAll(dest + dataTmpDir)
```

Comment:
should probably just make this deferred.

Reply:
Agreed

Reply:
if you defer this above the error checks, then you don't need to worry about it if any of the errors happen since it will get cleaned up anyway.